### DISCUSSION TITLE
Deprecation of Windows support

### DISCUSSION THREAD

**Discussion by chenbh** [buildpacks-community/kpack#1366] [General]
We (the kpack team at VMware) would like to deprecate support for Windows builds and want to gather some feedback before we start the formal process of submitting an RFC.

## Background

Kpack currently supports building Windows containers on Windows clusters. To do this, it has Windows specific code in the [build pod generator](https://github.com/buildpacks-community/kpack/blob/759bd290a5310bef5069385b649883e586024dae/pkg/apis/build/v1alpha2/build_pod.go) as well as shipping Windows variants of the [`build-init` and `completion` images](https://github.com/buildpacks-community/kpack/blob/759bd290a5310bef5069385b649883e586024dae/.github/workflows/ci.yaml#L151-L189) (these images are used in the build pod for pulling source code and signing built images respectively).

The supporting the Windows logic in the build pod generator isn't too difficult, but building and shipping the Windows images is a lot more complicated. 

## The Windows images

The source code for the 2 images are exactly the same as the linux images, `pkg/cmd/build-init/main.go` and `pkg/cmd/completion/main.go` respectively. The main problem comes with creating the builder image that is then used to build these 2 commands.

As far as I'm aware (would love to hear otherwise), there's currently no open source/publicly available CNB builder based on Windows. There's also no buildpack for golang that either natively runs on Windows, or cross compiles to Windows. To solve this problem, previous teams at VMware created a closed-sourced experimental buildpack based on `paketo-buildpack/go` that can natively run on Windows.  

This buildpack is then paired with couple of Dockerfiles pinned to a specific version of `mcr.microsoft.com/windows/nanoserver` plus some commands to copy DLLs, to create the stacks and eventually the builder image that is used in the kpack build process.

## Motivation for removal

1. We (VMware) aren't comfortable with open sourcing our experimental buildpack because we don't have enough experience with Windows to fully maintain and support it. And having a CNCF project built using some proprietary image is not a great process.
1. From our understanding, very small amount of people are actually using kpack on Windows clusters to build Windows images. It's simply not worth the maintainer overhead

## Discussion

Our question to the community is, are you using kpack to build Windows containers? If so how are you doing it and what's your use case?

**Comment by AidanDelaney** [comment]
I have no need of Windows support in kpack.  Furthermore, there is a draft buildpacks RFC to remove Windows support from `pack` and `lifecycle` https://github.com/buildpacks/rfcs/pull/311
